### ISSUE TITLE
Adding npm package as project dependency fails due to trace dependency

### ISSUE THREAD

**Issue by roben** [bootprint/bootprint#41]
Adding the npm package with node > 8 as local dependency fails with the following error:

```bash
yarn add --dev bootprint
yarn add v0.27.5
[1/4] Resolving packages...
warning bootprint > live-server > connect@2.30.2: connect 2.x series is deprecated
[2/4] Fetching packages...
error trace@2.4.1: The engine "node" is incompatible with this module. Expected version "^4.5 || ^6.0 || ^7.0".
error Found incompatible module
```

It still works fine, though, when adding it as global dependency which is our current workaround. 

As stated on `https://www.npmjs.com/package/trace`, "Trace only works with node.js v8.x and newer. Use npm install trace@^2 for node.js v6 and v4". So, updating the `trace` dependency should solve this issue.

(The same issue appears with `npm`, we just like to use `yarn`)

**Comment by igor-savin-ht** [comment]
One possible workaround for this would be to make trace an optional dependency via peerDependency.
trace-and-clarify-if-possible that is used by bootprint already handles case of running on Node 8 by skipping the trace, so not having trace installed is not going to break anything there, and whoever needs it on older Nodes can just install the dependency explicitly.

**Comment by igor-savin-ht** [comment]
@nknapp I can provide a PR for this if you agree with the approach.

**Comment by nknapp** [comment]
I would prefer this problem to be solved in the "trace-and-clarify-if-possible" package, because I created it exactly for this purpose: preventing errors sue to incompatible node versions.

We should discuss possible solutions there. If this is not possible at all, we can make it a peer dependency, but let's first try other approaches

**Comment by nknapp** [comment]
Resolved with 1.0.2 and `trace-and-clarify-if-possible@1.0.3`
